client/jetpack-connect/plans.jsx

```diff
@@ -50,38 +50,38 @@ class Plans extends Component {
 	redirecting = false;
 
 	componentDidMount() {
-		this.maybeRedirect( this.props );
+		this.maybeRedirect();
 		if ( ! this.redirecting ) {
 			this.props.recordTracksEvent( 'calypso_jpc_plans_view', {
 				user: this.props.userId,
 			} );
 		}
 	}
 
-	componentWillReceiveProps = nextProps => {
+	componentDidUpdate() {
 		if ( ! this.redirecting ) {
-			this.maybeRedirect( nextProps );
+			this.maybeRedirect();
 		}
-	};
+	}
 
-	maybeRedirect = props => {
-		if ( props.isAutomatedTransfer ) {
-			this.props.goBackToWpAdmin( props.selectedSite.URL + JETPACK_ADMIN_PATH );
+	maybeRedirect() {
```

Comment:
It could be moved, but I don't think that avoids any duplication.

Note that in `componentDidMount`, we first do `maybeRedirect`, which may result in `this.redirect === true`. Then we check `this.redirect` to record the analytics (or not).

Reply:
Which makes me think whether this is actually expected...why would want to `maybeRedirect` if we're currently redirecting?

Reply:
There's a few different ways to do this, none of which I like. I only reluctantly added the `redirect` flag back in after trying to do it all with props. There are just too many props in this component to do it simply.

Maybe the best way would be to check the flag immediately before setting the flag, right before we actually do a redirect. I think that would be best in another PR, though.

> Which makes me think whether this is actually expected...why would want to maybeRedirect if we're currently redirecting?

We should never be redirecting as we enter `componentDidMount()`, so we don't strictly need to check the flag. We check the flag on the way out of that method to see if we actually did redirect as a result of the props, in which case we don't want the `tracks` call.

This is tedious, but is trying to maintain the previous `tracks` behaviour.